pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -222,10 +223,13 @@ def precision_plugin(self) -> PrecisionPlugin:
 
     @property
     def training_type_plugin(self) -> TrainingTypePlugin:
+        if self._training_type_plugin_resolved:
+            # avoid calling `resolve_training_type_plugin` multiple times
```

Comment:
Thanks. Yes I fully understand the motivation of the PR and why you want to change it. There an be the inconsistency and it's great that you are improving it :)) 

So as you say, the reason why you implement the configuration here in the property getter is because you ran into problems with DDPSpawn? Is that correct?

Reply:
@awaelchli , I might missed your point.

for the 
> why do we "resolve" the plugin here and not directly where it is created/registered

i think it was "resolved" her also before (but it could actually been resolved multiple times, every time ), do you mind explaining a bit on where do you think we should "resolve".

> why you implement the configuration here in the property getter is because you ran into problems with DDPSpawn

well, actually not why implement here, you could say that is why I introduced this `self._training_type_plugin_resolved`. (with `DDPSpawn`, even with the same num_nodes, when resetting, it would cause some problem, I could try to repro the problem)

Reply:
> i think it was "resolved" her also before (but it could actually been resolved multiple times, every time ), do you mind explaining a bit on where do you think we should "resolve".

Yes you are right, I did not see that 😅 

Okay so I'm alright keeping it like this for now. But it looks to me like a source of error because what if the property is called here too early when not yet all information is known to resolve it.

Reply:
That is a valid point. I feel we should be explicit where this is resolved (after all required information is known), and only resolve only once (as `training_type_plugin` is a property, it could be accessed from other classes, exposing resolving logic inside might be not ideal)

I probably prefer the following:
```
....
self.handle_given_plugins()
self._training_type_plugin = self.select_training_type_plugin()
self._training_type_plugin = self.resolve_training_type_plugin(self._training_type_plugin)
self._precision_plugin = self.select_precision_plugin()
self.accelerator = self.select_accelerator()
...
```
for the property, change to

```
    @property
    def precision_plugin(self) -> PrecisionPlugin:
        assert self._precision_plugin is not None
        return self._precision_plugin

    @property
    def training_type_plugin(self) -> TrainingTypePlugin:
        assert self._training_type_plugin is not None
        return self._training_type_plugin
```